skyVegasTests/gamePlayGameClient.spec.ts

```diff
@@ -1,35 +1,205 @@
-import { test, type Page } from '@playwright/test';
+import { test, type Page, expect } from '@playwright/test';
 import {
     launchGame,
     login,
-    validateConsoleMessages,
     startEventListener,
-    makeSpin,
+    firstSpin,
     readGames,
+    spin,
+    getValueFromConsoleMessages,
+    getBalanceGameWindow,
+    getStakeAmountGameWindow,
+    getWinAmountGameWindow,
+    getWinLossAmountGameWindow,
 } from './utilities';
 import { ExpectedMessage } from '../types/expectedMessage';
+import { numberToTwoDecimalPlaces } from './utilities/getValuesFromConsoleMessagesHelper';
 
 let page: Page;
 readGames().forEach((game) => {
-    const consoleMessages: string[] = [];
-
     test.describe(`Testing with text: ${game}`, () => {
-        test.beforeAll(async ({ browser }) => {
+        test.beforeEach(async ({ browser }) => {
+            test.setTimeout(300000);
             page = await browser.newPage();
             await login(page);
         });
 
-        test.afterAll(async () => {
+        test.afterEach(async () => {
             await page.close();
         });
 
-        test('gamePlay', async () => {
+        test('gamePlayForLoss', async () => {
+            const consoleMessages: string[] = [];
+            let numberOfSpins = 0;
             startEventListener(page, consoleMessages);
+
             await launchGame(page, game, consoleMessages);
-            await makeSpin(page, consoleMessages);
-            await validateConsoleMessages(
-                ExpectedMessage.END_SPIN,
-                consoleMessages
+
+            const startBalanceConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.BALANCE_UPDATE
+            );
+
+            const startBalanceGameWindow = await getBalanceGameWindow(page);
```

Comment:
We have reported bugs with providers where these values were not populated on the game window. Would be good to add assertions that the values exist along with the comparison assertions we have at the moment.

Reply:
Good shout, will add this in !